### PR TITLE
random: add sys_rand8_get() and sys_rand16_get()

### PR DESCRIPTION
**Pull request by maass-hamburg**
add sys_rand8_get() and sys_rand16_get()

### CODE REVIEW

File: subsys/random/rand32_entropy_device.c [hunk 1]
```diff
@@ -13,6 +13,52 @@ static const struct device *const entropy_dev =
 	DEVICE_DT_GET(DT_CHOSEN(zephyr_entropy));
 
 #if defined(CONFIG_ENTROPY_DEVICE_RANDOM_GENERATOR)
+uint8_t z_impl_sys_rand8_get(void)
+{
+	uint8_t random_num;
```
Comment (jukkar) [inline subsys/random/rand32_entropy_device.c:18]: Would it be simpler to do this here, or is it important to use smaller number of bytes from entropy?
```
return (uint8_t)sys_rand32_get();
```
and similar comment to 16-bit version.

Comment (maass-hamburg) [reply subsys/random/rand32_entropy_device.c:18]: there it is important, explicitly for external entropy sources, which are connected via I2C or SPI, resources could be saved. For example the driver of my PR #69920 would benefit from it.

File: subsys/random/rand32_timer.c [hunk 2]
```diff
@@ -47,6 +47,16 @@ uint32_t z_impl_sys_rand32_get(void)
 	return val;
 }
 
+uint8_t z_impl_sys_rand8_get(void)
+{
+	return (uint8_t)z_impl_sys_rand32_get();
```
Comment (jukkar) [inline subsys/random/rand32_timer.c:52]: Shouldn't you call `z_impl_sys_rand8_get()` here?
And similar comment to rand16 below.

File: subsys/random/rand32_xoshiro128.c [hunk 3]
```diff
@@ -100,6 +100,16 @@ uint32_t z_impl_sys_rand32_get(void)
 	return xoshiro128_next();
 }
 
+uint8_t z_impl_sys_rand8_get(void)
+{
+	return (uint8_t)z_impl_sys_rand32_get();
```
Comment (jukkar) [inline subsys/random/rand32_xoshiro128.c:105]: ditto

File: subsys/random/rand32_timer.c (hunk 2 above)
Comment (maass-hamburg) [reply subsys/random/rand32_timer.c:52]: for the timer we use the implementation for 32 bit as it wouldn't be that much more efficient to have a custom implementation for 8 bit as it also would have to use `k_cycle_get_32()`. The aim of this PR is to optimise the use of the entropy device. The other entropy sources also get a implementation for 8 and 16 bit, but these will use the existing implementation of `sys_rand32_get()`, to be compatible.

File: subsys/random/rand32_xoshiro128.c (hunk 3 above)
Comment (maass-hamburg) [reply subsys/random/rand32_xoshiro128.c:105]: see above

**Comment by ceolin** [comment]
Not against adding helpers for `uint8_t` and `uint16_t`, but the way it is implemented is adding a lot of duplicated code. That is not an exclusive problem of this pr. I have simplified it in https://github.com/zephyrproject-rtos/zephyr/pull/70695

Please take a look there and see if we can make these changes on top of that.

**Review by ceolin — CHANGES_REQUESTED** [review]
-1 due the amount of duplicated code. Please take a look in the other comment.

**Comment by maass-hamburg** [comment]
> Not against adding helpers for `uint8_t` and `uint16_t`, but the way it is implemented is adding a lot of duplicated code. That is not an exclusive problem of this pr. I have simplified it in #70695
> 
> Please take a look there and see if we can make these changes on top of that.

@ceolin it now follows the implementation of `sys_rand32_get()` from https://github.com/zephyrproject-rtos/zephyr/pull/70695

**Review by jukkar — DISMISSED** [review]

**Comment by maass-hamburg** [comment]
rebased to resolve conflicts

**Review by ceolin — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Pull request merged**